File: itmd-521/labs/module-3/mnmcount.py

import sys
from pyspark.sql import SparkSession
from pyspark.sql.functions import col, count

spark = SparkSession.builder.appName("PythonMnMCount").getOrCreate()

mnm_file = "/home/vagrant/LearningSparkV2/databricks-datasets/learning-spark-v2/mnm_dataset.csv"

mnm_df = spark.read.format("csv")
         .option("header", "true")
         .option("inferSchema", "true")
         .load(mnm_file)

count_mnm_df = (
    mnm_df
    .groupBy("State", "Color")
    .agg(count("Count").alias("Total"))
    .orderBy(col("Total").desc())
)

count_mnm_df.show(8)

spark.stop()

File: itmd-521/labs/module-3/mnmcount.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._

object MnMCount {
  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder
      .appName("MnMCount")
      .getOrCreate()

    val mnmFile = "/home/vagrant/LearningSparkV2/databricks-datasets/learning-spark-v2/mnm_dataset.csv"

    val mnmDF = spark.read
      .option("header", "true")
      .option("inferSchema", "true")
      .csv(mnmFile)

    val countMnMDF = mnmDF
      .groupBy("State", "Color")
      .agg(count("Count").alias("Total"))
      .orderBy(col("Total").desc)

    countMnMDF.show(8)

    spark.stop()
  }
}
